caffe2/python/pybind_state_dlpack.h

```diff
@@ -28,7 +28,7 @@ class DLPackWrapper {
       : tensor(tensor), device_option(device_option) {}
 
   py::object data() {
-    DLContext tensor_context;
+    DLDevice tensor_context;
```

Comment:
The internal error is

```
caffe2/python/pybind_state_dlpack.h:31:5: error: unknown type name 'DLDevice'; did you mean 'Device'?
    DLDevice tensor_context;
```

which I don't understand.

@t-vi, what would happen if we didn't update Caffe2 and just updated ATen/dlpack.h?

Reply:
My interest is exclusively with PyTorch /ATen, so I neither know nor could I say whether any particular outcome was good or bad. I was more concerned that you might have internal interfaces between them and caffe2.

Reply:
I removed the changes in caffe2/ (and rebased to master from a few days ago).